service/common/src/main/java/org/apache/polaris/service/exception/IcebergExceptionMapper.java

```diff
@@ -191,6 +192,7 @@ static int mapExceptionToResponseCode(RuntimeException rex) {
       case IllegalArgumentException e -> Status.BAD_REQUEST.getStatusCode();
       case UnsupportedOperationException e -> Status.NOT_ACCEPTABLE.getStatusCode();
       case WebApplicationException e -> e.getResponse().getStatus();
+      case PersistenceException e -> Status.UNAUTHORIZED.getStatusCode();
```

Comment:
What's the rationale for mapping `PersistenceException` to `UNAUTHORIZED`? Are we sure all of those errors are caused by failed authorization checks?

Reply:
Hmm, looks like 500 is more suitable for `PersistenceException`. The 401 status code is reserved for authentication failures, which isn't applicable here. We may only need to mask the sensitive information a bit.

Reply:
Thank you for your feedback 🙏 
I've adjusted this PR and changed the error code back to 500